hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java

```diff
@@ -194,13 +194,17 @@ private boolean isDeletionAllowed(ContainerData containerData,
       return false;
     } else if (!containerData.isClosed()) {
       return false;
+    } else if (containerData.getOriginPipelineId() == null
```

Comment:
what if a ratis container does not have a originPipelineID exceptionally? is it a good behavior to just return true directlly?

IMHO, the problem here is caused by 
`if (ozoneContainer.getWriteChannel() instanceof XceiverServerRatis)`.
we should not differentiate containers by the write channel , we should differentiate them by container type. so the code should be like
```
ContainerType contianerType = containerData.getContainerType();
switch(contianerType) {
case Ratis:
return isDeletionAllowedForRatis()
default:
//for non-ratis container, they do not have pipelineid
return true:
}
```
what is more, we could also add a sanity check for ratis container pipeline in `isDeletionAllowedForRatis()` to make sure the pipelineid is not empty

Reply:
> what if a ratis container does not have a originPipelineID exceptionally? is it a good behavior to just return true directlly?

If a ratis container does not have a originPipelineID, how do we get the `minReplicatedIndex` then?

> we should not differentiate containers by the write channel , we should differentiate them by container type.

I agree with this. But it should be done in another PR.